frontend/src/service/apiService.ts

```diff
@@ -0,0 +1,27 @@
+export const get = async (url: String, token: String | undefined) => {
+  const response = await fetch(url as RequestInfo, {
+    headers: { Authorization: `Bearer ${token}` },
+  });
+  return response;
+};
```

Comment:
I'm going to be a pain in the neck, but what happens if the GET call fails for whatever reason?
I think we need a way to handle errors (which I haven't done previously myself, so who am I to criticise).
You can add `.catch(callbackFunction)` to `await fetch`, but then the error needs to be propagated out of the function.

There are two ways I can think of:
1. change the function signature to accept an error callback function that will tell you what to do if an error is encountered.
2. we can stick to the functional programming approach and use e.g. `Result` from `folktale` package https://folktale.origamitower.com/api/v2.3.0/en/folktale.result.html - the function will then return either `Ok(response)` or `Err(error)`.

The latter option looks nicer to me, but I'm happy with either - we do need some feedback about why a call fails either way.

Reply:
I wasn't sure if it was best to have it here or in the thunk itself. I felt like I was repeating myself in `staffSlice`. You had a nice function in there before which could adapt to the different methods, I'm sure it could be neatened up a bit?

Reply:
I think this is better the way you have it now. It allows us to use the api calls without invoking redux. There is some repetition both here and in staffSlice, but you can avoid it by writing some helper functions, e.g. here:
```
const fetchUrl = (method: string) => async (url: string, token?: string, body?: string) => 
    await fetch(url as RequestInfo, {
      headers: { Authorization: `Bearer ${token}` },
      method,
      body,
  });

export const get = fetchUrl('GET');
```

and in staffSlice by passing the function as an argument. 
It's a bit cleaner, but not a big deal imo.

Reply:
What would happen in a case where body isn't needed for example in a GET request? Can body be left as undefined?

Reply:
I think you can just omit it and it will be left as `undefined`. Typescript won't complain, because it's declared as an optional argument.
The only potential problem is if you want body but not the token, I think you'd have to do
`fetchUrl('url', undefined, 'token')`.

I think it should be fine, but it should be verified.

Reply:
Ah yeah I see. I think for now I will leave it as it is and keep the method invocations simple but we can easily change this down the line!